### DISCUSSION TITLE
Issues about freeimage library

### DISCUSSION THREAD

**Discussion by topazus** [linuxdeepin/developer-center#7740] [Developer Discussion | 开发者讨论]
I received an email about the plan of orphaning freeimage package on Fedora from its package maintainer:

> I intend to orphan freeimage. Probably, the package should rather just
> be retired. Upstream is effectively dead, and there is a constant stream
> of CVEs getting filed against the package which are not addressed
> upstream. Over the past two years I've fixed many of those CVEs
> downstream, but the most recent batch of 15 CVEs is leading me to
> capitulate. Currently, the following packages require freeimage:
> 
> PerceptualDiff
> allegro5
> cegui06
> deepin-image-viewer
> gazebo
> imv
> ogre
> photoqt
> 
> A minimal impact check:
> 
> PerceptualDiff: freeimage is a hard dependency. PerceptualDiff itself
> has seen its last commit 4 years ago, last release 8 years ago
> allegro5: freeimage is an optional dependency
> cegui06: freeimage is an optional dependency
> deepin-image-viewer: freeimage is a hard dependency
> gazebo: freeimage is a hard dependency. (The fedora package is for
> "gazebo-classic" https://github.com/gazebosim/gazebo-classic, which
> points to https://github.com/gazebosim/gz-sim as the "latest version",
> which does not appear to require freeimage)
> imv: freeimage is an optional dependency
> ogre: freeimage is an optional dependency
> photoqt: freeimage is an optional dependency

There are several issues that need to be solved by freeimage. see details: https://github.com/arrayfire/forge/issues/248. Any suggestions?

**Comment by UTsweetyfish** [comment]
Hi topazus,

Currently deepin-image-viewer heavily relies on freeimage, and we do not have plans to change this dependency now.
While we may consider alternatives in the future, we cannot make any promises at this time.

**Comment by UTsweetyfish** [comment]
deepin-image-viewer has dropped dependency on freeimage.
